src/v3/src/components/PasswordWithConfirmation/PasswordWithConfirmation.tsx

```diff
@@ -0,0 +1,135 @@
+import { h } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+
+import { getMessage } from '../../../../v2/ion/i18nTransformer';
+import { useWidgetContext } from '../../contexts';
+import { useValue } from '../../hooks';
+import {
+  ChangeEvent,
+  DataSchema,
+  FormBag,
+  PasswordWithConfirmationElement,
+  UISchemaElementComponent,
+} from '../../types';
+import InputPassword from '../InputPassword';
+
+const PasswordWithConfirmation: UISchemaElementComponent<{
+  uischema: PasswordWithConfirmationElement
+}> = ({ uischema }) => {
+  const { input, fieldRequiredErrorMessage, passwordMatchErrorMessage } = uischema.options;
+  const {
+    inputMeta: {
+      // @ts-ignore expose type from auth-js
+      messages: newPasswordMessages = {},
+    },
+  } = input.options;
+  const newPasswordError = newPasswordMessages?.value?.[0]
+    && getMessage(newPasswordMessages.value[0]);
+
+  const value = useValue(input);
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+  const [confirmPassword, setConfirmPassword] = useState<string>();
+  const [confirmPasswordError, setConfirmPasswordError] = useState<string | undefined>(
+    fieldRequiredErrorMessage,
+  );
+  const { dataSchemaRef } = useWidgetContext();
+
+  const handleConfirmPasswordValidation = (password: string | undefined): void => {
+    if (!password) {
+      setConfirmPasswordError(fieldRequiredErrorMessage);
+      return;
+    }
+
+    if (value !== password) {
+      setConfirmPasswordError(passwordMatchErrorMessage);
+      return;
+    }
+
+    setConfirmPasswordError(undefined);
+  };
+
+  const updateDataSchemaValidation = (confirmPw?: string): void => {
+    dataSchemaRef.current![input.name] = {
+      validate: (data: FormBag['data']) => {
+        const newPw = data[input.name];
+        const comparisonPw = (confirmPw ?? confirmPassword);
+        if (!newPw) {
+          setIsTouched(true);
+          return { i18n: { key: 'model.validation.field.blank' } };
+        }
+
+        if (comparisonPw !== newPw) {
+          setIsTouched(true);
+          // Do not display error on new password field when confirm is missing
+          // but do not allow submission
+          return { i18n: { key: '' } };
+        }
+
+        return undefined;
+      },
+    } as DataSchema;
+  };
+
+  const handleConfirmPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setIsTouched(true);
+    setConfirmPassword(e.currentTarget.value);
+    updateDataSchemaValidation(e.currentTarget.value);
+    handleConfirmPasswordValidation(e.currentTarget.value);
+  };
+
+  const handleConfirmPasswordBlur = () => {
+    setIsTouched(true);
+    updateDataSchemaValidation();
+    handleConfirmPasswordValidation(confirmPassword);
+  };
+
+  useEffect(() => {
+    // on load, update validate function to prevent submission w/o confirm pw data
+    updateDataSchemaValidation();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [newPasswordError]);
```

Comment:
It is because messages are pushed down to components via the inputs field. When you submit with field level errors, the validate function gets reset back to the original value, this needs to know when that happens so we can overwrite the default validate function whenever that happens.